app/components/Tables/FullTable.tsx

```diff
@@ -125,6 +128,10 @@ function FullTable<Type>({
     }
   }, [])
 
```

Comment:
What is the purpose of the useState inside the useEffect?

Normally this is an indicator that something needs to be refactored to avoid this from happening. As doing this can be the cause of rerender loops.

Reply:
Hi, I thought that this was a good way to keep the globalFilter state in sync with the filter prop. Looking at it more, one fix could be to pass the filter prop directly into the state option in useReactTable and get rid of the useState. Maybe something like this ```state: {
      columnFilters,
      globalFilter: filter,
      columnOrder
    }```? Also, I could change the prop name from filter to globalFilter to keep the state option as ```state: {
      columnFilters,
      globalFilter,
      columnOrder
    }```.

Reply:
Yeah, alternatively you can move the state to the parent component and then this component will sort itself out each time the filter changes. You can then pass through the state setter function as a prop too.